src/actions/catalogActions.js

```diff
@@ -5,37 +5,29 @@ import { StatusCodes } from 'shared/constants';
 import * as types from './actionTypes';
 
 // loadCatalog takes a catalog object and tries to load further data.
-function loadCatalogIndex(catalog) {
-  return fetch(`${catalog.spec.storage.URL}index.yaml`, { mode: 'cors' })
-    .catch(() => {
-      return fetch(
-        `https://cors-anywhere.herokuapp.com/${catalog.spec.storage.URL}index.yaml`,
-        { mode: 'cors' }
-      );
-    })
-    .catch((error) => {
-      throw error;
-    })
-    .then((response) => {
-      if (response.status === StatusCodes.Ok) {
-        return response.text();
-      }
-
-      throw new Error(
-        `Could not fetch index.yaml at ${catalog.spec.storage.URL}`
-      );
-    })
-    .then((body) => {
-      const rawCatalog = yaml.safeLoad(body);
-      const catalogWithApps = Object.assign({}, catalog, {
-        apps: rawCatalog.entries,
-      });
+async function loadCatalogIndex(catalog) {
+  let indexURL = `${catalog.spec.storage.URL}index.yaml`;
 
-      return catalogWithApps;
-    })
-    .catch((error) => {
-      throw error;
-    });
+  if (indexURL.includes('kubernetes-charts.storage.googleapis.com')) {
```

Comment:
I know it's a longshot but could this be tightened to enforce the root domain? E.g. to prevent things like `kubernetes-charts.storage.googleapis.com.my.evil.site`

Reply:
Absolutely! I'll change this to `indexURL === "https://kubernetes-charts.storage.googleapis.com/index.yaml"`